tests/v2/validation/observability/stackstate_server_test.go

```diff
@@ -0,0 +1,271 @@
+
+	sss.catalogClient, err = sss.client.GetClusterCatalogClient(sss.cluster.ID)
+	require.NoError(sss.T(), err)
+
+	projectTemplate := kubeprojects.NewProjectTemplate(cluster.ID)
+	projectTemplate.Name = charts.StackstateNamespace
+	project, err := client.Steve.SteveType(project).Create(projectTemplate)
+	require.NoError(sss.T(), err)
+	sss.projectID = project.ID
+
+	ssNamespaceExists, err := namespaces.GetNamespaceByName(client, cluster.ID, charts.StackstateNamespace)
+	if ssNamespaceExists == nil && k8sErrors.IsNotFound(err) {
+		_, err = namespaces.CreateNamespace(client, cluster.ID, project.Name, charts.StackstateNamespace, "", map[string]string{}, map[string]string{})
+	}
+	require.NoError(sss.T(), err)
+
+	var stackstateConfigs observability.StackStateConfig
+	config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+	sss.stackstateConfigs = &stackstateConfigs
+
+	// Install StackState Chart Repo
+	_, err = sss.catalogClient.ClusterRepos().Get(context.TODO(), observabilityChartName, meta.GetOptions{})
+	if k8sErrors.IsNotFound(err) {
+		err = charts.CreateClusterRepo(sss.client, sss.catalogClient, observabilityChartName, observabilityChartURL)
+		log.Info("Created suse-observability repo StackState install.")
+	}
+	require.NoError(sss.T(), err)
+
+	latestSSVersion, err := sss.catalogClient.GetLatestChartVersion(charts.StackStateChartRepo, observabilityChartName)
+
+	sss.stackstateChartInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestSSVersion,
+		ProjectID: sss.projectID,
+	}
+}
+
+func (sss *StackStateServerTestSuite) TestInstallStackState() {
+	subsession := sss.session.NewSession()
+	defer subsession.Cleanup()
+
+	sss.Run("Install SUSE Observability Chart", func() {
+		// Read cattle config
+		var stackstateConfigs observability.StackStateConfig
+		config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+
+		// Set base config values
+		baseConfig := observability.BaseConfig{
+			Global: struct {
+				ImageRegistry string `json:"imageRegistry" yaml:"imageRegistry"`
+			}{
+				ImageRegistry: "registry.rancher.com",
+			},
+			Stackstate: observability.StackstateServerConfig{
+				BaseUrl: stackstateConfigs.Url,
+				Authentication: observability.AuthenticationConfig{
+					AdminPassword: stackstateConfigs.AdminPassword,
+				},
+				ApiKey: observability.ApiKeyConfig{
+					Key: stackstateConfigs.ClusterApiKey,
+				},
+				License: observability.LicenseConfig{
+					Key: stackstateConfigs.License,
+				},
+			},
+		}
+
+		ingressConfig := observability.IngressConfig{
+			Ingress: observability.Ingress{
+				Enabled: true,
+				Annotations: map[string]string{
+					"nginx.ingress.kubernetes.io/proxy-body-size": "50m",
+				},
+				Hosts: []observability.Host{
+					{
+						Host: stackstateConfigs.Url,
+					},
+				},
+				TLS: []observability.TLSConfig{
+					{
+						Hosts:      []string{stackstateConfigs.Url},
+						SecretName: "tls-secret",
+					},
+				},
+			},
+		}
```

Comment:
the structure of this is a little confusing to me, why is there a config that is setup to be used vs. making this part of the functions directly in actions/charts/stackstate.go? might be worth a huddle if you have time. But iirc if you do it in the chart + get the default values first, then override the vars you send along with the create, you should be able to avoid these map/interface conversions